src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -208,10 +208,13 @@ impl<'a, 'b:'a, 'tcx:'b> GraphBuilder<'a, 'b, 'tcx> {
                     // Return an error here by looking up the namespace that
                     // had the duplicate.
                     let ns = ns.unwrap();
-                    self.resolve_error(sp,
-                        &format!("duplicate definition of {} `{}`",
-                             namespace_error_to_string(duplicate_type),
-                             token::get_name(name)));
+                    ::resolve_error(
+                        self,
+                        sp,
+                        &::ResolutionError::DuplicateDefinition(
```

Comment:
There's no need to pass this by reference AFAICT. Can also import `ResolutionError` to get rid of the absolute path's leading `::`.